tests/ui/traits/new-solver/object-unsafety.stderr

```diff
@@ -14,6 +14,12 @@ help: consider introducing a `where` clause, but there might be an alternative b
 LL | pub fn copy_any<T>(t: &T) -> T where dyn Setup<From = T>: Setup {
    |                                ++++++++++++++++++++++++++++++++
 
-error: aborting due to previous error
+error[E0277]: the trait bound `for<'a> fn(&'a <dyn Setup<From = T> as Setup>::From) -> <dyn Setup<From = T> as Setup>::From {copy::<dyn Setup<From = T>>}: Callable<(&T,)>` is not satisfied
+  --> $DIR/object-unsafety.rs:12:5
+   |
+LL |     copy::<dyn Setup<From = T>>(t)
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Callable<(&T,)>` is not implemented for fn item `for<'a> fn(&'a <dyn Setup<From = T> as Setup>::From) -> <dyn Setup<From = T> as Setup>::From {copy::<dyn Setup<From = T>>}`
```

Comment:
Not exactly sure what happened here, but I'm assuming we just don't explain that `Callable` isn't implemented because `Setup` isn't implemented for `dyn Setup`, but just fail the outer obligation.

Orthogonally to that I can make it less confusing by adding a `rustc_on_unimplemented` to `Callable` to give a more targeted message.

Reply:
It's because `&'a <dyn Setup<From = T> as Setup>::From` isn't normalizing to `&T`.